src/test/java/karuna/karuna_backend/Services/UserServiceTest.java

```diff
@@ -0,0 +1,91 @@
+package karuna.karuna_backend.Services;
+
+import karuna.karuna_backend.DTO.UserDTO;
+import karuna.karuna_backend.Models.User;
+import karuna.karuna_backend.Repositories.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    @Test
+    public void testGetUserById_Found() {
+        // Mock data
+        int userId = 1;
+        String username = "test_user";
+        String password = "test_password";
+        User user = new User(userId, username, password);
+        Optional<User> optionalUser = Optional.of(user);
+
+        // Mock repository behavior
+        when(userRepository.findByID(userId)).thenReturn(optionalUser);
+
+        // Call the method
+        Optional<UserDTO> userDTO = userService.getUserById(userId);
+
+        // Assertions
+        assertTrue(userDTO.isPresent());
+        assertEquals(username, userDTO.get().getUsername());
```

Comment:
Here is a thing, every unit test should check for only one think. So if you want to create 2 assertions in 1 Test in that case you can do this easily 1 lane and it does not matter. However If you want to check 2 things, first is that user exists and second his nickname create 2 separate @Test methods and then move whole setup to @Before or it's derivatives functions. I understand it was GPT printed because no need to manually write simple tests tbh but always try to thing if there is something better to do. To summarize everything up ->
 split this into 2 separate test methods.